# Basic Statistics Assignment

In [1]:
import statsmodels.api as sm
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

/usr/local/lib/python3.7/dist-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
  import pandas.util.testing as tm


In [3]:
df = pd.read_csv('https://github.com/kaopanboonyuen/2110446_DataScience_2021s2/raw/main/datasets/adult.data.csv')
df.head()

age         workclass  fnlwgt  ... hours-per-week  native-country salary
0   39         State-gov   77516  ...             40   United-States  <=50K
1   50  Self-emp-not-inc   83311  ...             13   United-States  <=50K
2   38           Private  215646  ...             40   United-States  <=50K
3   53           Private  234721  ...             40   United-States  <=50K
4   28           Private  338409  ...             40            Cuba  <=50K

[5 rows x 15 columns]

In [19]:
df.tail()

age     workclass  fnlwgt  ... hours-per-week  native-country salary
32556   27       Private  257302  ...             38   United-States  <=50K
32557   40       Private  154374  ...             40   United-States   >50K
32558   58       Private  151910  ...             40   United-States  <=50K
32559   22       Private  201490  ...             20   United-States  <=50K
32560   52  Self-emp-inc  287927  ...             40   United-States   >50K

[5 rows x 15 columns]

![alt text](https://media.giphy.com/media/L20mbc7yRfsly/giphy.gif)



```
`# This is formatted as code`
```

## Part 1: EDA

In [7]:
# 1. How many men and women (sex feature) are represented in this dataset?

# You code here
print(df[df.sex == "Male"].sex.count())
print(df[df.sex == "Female"].sex.count())

21790
10771


In [14]:
# 2. What is the average, min, max, std, and median age (age feature) of women?

# You code here
df[df.sex == "Female"].age.describe()

count    10771.000000
mean        36.858230
std         14.013697
min         17.000000
25%         25.000000
50%         35.000000
75%         46.000000
max         90.000000
Name: age, dtype: float64

In [18]:
# 3. What is the percentage of United-States citizens (native-country feature)?

# You code here
df[df["native-country"] == "United-States"].count() / df.count() * 100

age               89.585701
workclass         89.585701
fnlwgt            89.585701
education         89.585701
education-num     89.585701
marital-status    89.585701
occupation        89.585701
relationship      89.585701
race              89.585701
sex               89.585701
capital-gain      89.585701
capital-loss      89.585701
hours-per-week    89.585701
native-country    89.585701
salary            89.585701
dtype: float64

In [30]:
# 4. What are the mean and standard deviation of age for those who earn more than 50K per year (salary feature) 

# You code here
df[df.salary == ">50K"].age.describe()

count    7841.000000
mean       44.249841
std        10.519028
min        19.000000
25%        36.000000
50%        44.000000
75%        51.000000
max        90.000000
Name: age, dtype: float64

In [31]:
# From 4. what are the mean and standard deviation of age for those who earn less than 50K per year?

# You code here
df[df.salary == "<=50K"].age.describe()

count    24720.000000
mean        36.783738
std         14.020088
min         17.000000
25%         25.000000
50%         34.000000
75%         46.000000
max         90.000000
Name: age, dtype: float64

In [32]:
# 6. Is it true that people who receive more than 50k have at least high school education? 
# (education - Bachelors, Prof-school, Assoc-acdm, Assoc-voc, Masters or Doctorate feature)

# You code here
df[df.salary == ">50K"].education.unique()

array(['HS-grad', 'Masters', 'Bachelors', 'Some-college', 'Assoc-voc',
       'Doctorate', 'Prof-school', 'Assoc-acdm', '7th-8th', '12th',
       '10th', '11th', '9th', '5th-6th', '1st-4th'], dtype=object)

In [33]:
# 7. What is the maximum number of hours a person works per week (hours-per-week feature)?

# You code here
df["hours-per-week"].max()

99

In [45]:
# 8. How many people work such a number of hours and what is the percentage of those who earn a lot among them?

# You code here
df[(df["hours-per-week"] == 99) & (df.salary == ">50K")].count() / df[df["hours-per-week"] == 99].count()

age               0.294118
workclass         0.294118
fnlwgt            0.294118
education         0.294118
education-num     0.294118
marital-status    0.294118
occupation        0.294118
relationship      0.294118
race              0.294118
sex               0.294118
capital-gain      0.294118
capital-loss      0.294118
hours-per-week    0.294118
native-country    0.294118
salary            0.294118
dtype: float64

## Part 2: Two sample t-test

Is there a significant difference between the population proportions of parents of black children and parents of Hispanic children who report that their child has had some swimming lessons?

**Populations**: All parents of black children age 6-18 and all parents of Hispanic children age 6-18  
**Parameter of Interest**: p1 - p2, where p1 = black and p2 = hispanic  

**Null Hypothesis:** p1 - p2 = 0  
**Alternative Hypthosis:** p1 - p2 $\neq$ = 0  

**Data**: 247 Parents of Black Children. 36.8% of parents report that their child has had some swimming lessons. 
<br>308 Parents of Hispanic Children. 38.9% of parents report that their child has had some swimming lessons.

> Use of `ttest_ind()` from `statsmodels`
Difference in population proportion needs t-test. Also, the population follow a binomial distribution here. We can just pass on the two population quantities with the appropriate binomial distribution parameters to the t-test function.

The function returns three values: (a) test statisic, (b) p-value of the t-test, and (c) degrees of freedom used in the t-test.

In [75]:
# You code here
from scipy import stats

#                         val  prob  count
data1 = np.random.binomial(1, 0.368, 247)
data2 = np.random.binomial(1, 0.389, 308)

stats.levene(data1, data2)
statistic, pvalue = stats.ttest_ind(data1, data2, equal_var=True)

if (pvalue < 0.05):
  print(f"pvalue = {pvalue} reject") 
else:
  print(f"pvalue = {pvalue} accept")

0.8583923953209618
accept


## Part 3: One sample t-test

Let's say a cartwheeling competition was organized for some adults. The data looks like following,

(80.57, 98.96, 85.28, 83.83, 69.94, 89.59, 91.09, 66.25, 91.21, 82.7 , 73.54, 81.99, 54.01, 82.89, 75.88, 98.32, 107.2 , 85.53, 79.08, 84.3 , 89.32, 86.35, 78.98, 92.26, 87.01)

Is the average cartwheel distance (in inches) for adults more than 80 inches?

**Population**: All adults  
**Parameter of Interest**: $\mu$, population mean cartwheel distance.

**Null Hypothesis:** $\mu$ = 80 
<br>**Alternative Hypthosis**: $\mu$ > 80

**Data**:
<br>25 adult participants. 
<br>$\bar{x} = 83.84$
<br>$sd = 10.72$

In [81]:
# You code here

data = [80.57, 98.96, 85.28, 83.83, 69.94, 89.59, 91.09, 66.25, 91.21, 82.7 , 73.54, 81.99, 54.01, 82.89, 75.88, 98.32, 107.2 , 85.53, 79.08, 84.3 , 89.32, 86.35, 78.98, 92.26, 87.01]

statistic, pvalue = stats.ttest_1samp(data, 80)

if (pvalue < 0.05):
  print(f"pvalue = {pvalue} reject")
else:
  print(f"pvalue = {pvalue} accept")

pvalue = 0.09167485698904507 accept
